src/unstructured.rs

```diff
@@ -319,18 +319,24 @@ impl<'a> Unstructured<'a> {
             return Ok((start, 0));
         }
 
-        let delta = end - start;
-        debug_assert_ne!(delta, T::ZERO);
+        // From here on out we work with the unsigned representation. All of the
+        // operations performed below work out just as well whether or not `T`
+        // is a signed or unsigned integer.
+        let start = start.to_unsigned();
```

Comment:
Won't this fail for negative numbers?

Reply:
The `to_unsigned` conversion is implemented with `as` so this call will not fail for negative numbers. Happy to rename the method to `as_unsigned` or something, but also not sure what we would call the `Int::from_unsigned` method in that case.

Reply:
(See also the new test and fuzz target for signed integers and `int_in_range` which exercises this case)

Reply:
Ah, I didn't realize `as` had those semantics

Reply:
yeah it is just a bitcast for e.g. `u8` to `i8`